go-controller/pkg/ovn/endpoints.go

```diff
@@ -93,16 +90,34 @@ func (ovn *Controller) AddEndpoints(ep *kapi.Endpoints) error {
 			}
 			vip := util.JoinHostPortInt32(svc.Spec.ClusterIP, svcPort.Port)
 			ovn.AddServiceVIPToName(vip, svcPort.Protocol, svc.Namespace, svc.Name)
-			ovn.handleExternalIPs(svc, svcPort, lbEps.IPs, lbEps.Port, false)
+			if len(svc.Spec.ExternalIPs) > 0 {
+				gateways, _, err := ovn.getOvnGateways()
+				if err != nil {
+					return err
+				}
+				for _, gateway := range gateways {
+					loadBalancer, err := ovn.getGatewayLoadBalancer(gateway, svcPort.Protocol)
+					if err != nil {
+						klog.Errorf("Gateway router %s does not have load balancer (%v)", gateway, err)
+						continue
+					}
+					if loadBalancer == "" {
+						continue
+					}
+					if err = ovn.createLoadBalancerVIPs(loadBalancer, svc.Spec.ExternalIPs, svcPort.Port, lbEps.IPs, lbEps.Port); err != nil {
+						klog.Errorf("Error in creating Cluster IP for svc %s, target port: %d - %v\n", svc.Name, lbEps.Port, err)
+					}
```

Comment:
L108: The error message is not correct. Perhaps include the ExternalIP:Port in the error message as well.

Reply:
I've fixed the message, but I didn't add the ExternalIP:Port as we pass a list of external IPs to `createLoadBalancerVIPs` and in case something fails: the nested error should "bubble up" with the VIP message.